### PR TITLE
Auto Balance code integrated into smokey

### PR DESCRIPTION
**Pull request by josieammer**
Tested in lab on veer chassis.  Successfully balances ramp in a variety of stating conditions.

### CODE REVIEW

File: smokey/src/main/java/frc/robot/commands/AutoBalanceStepCommand.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    // do not need to add Navx as a requirement because it is read-only
+    addRequirements(drivetrainSubsystem);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    System.out.println("initializaing auto balance step command");
+    //Translation2d angleOfSteepestAscent = QuaternionUtils.getAngleOfSteepestAscent(navxsubsystem.getQuaterion());
```
Comment (mmcadamsms) [inline smokey/src/main/java/frc/robot/commands/AutoBalanceStepCommand.java:53]: remove the commented out stuff ... I presume we don't need it?

File: smokey/src/main/java/frc/robot/commands/AutoBalanceStepCommand.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    driveTimer.start();
+    done = false;
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute()
+  {
+    // drive along the vector of steepest ascent
+    drivetrainsubsystem.drive(new ChassisSpeeds(xVelocity, yVelocity, rotVelocity));
```
Comment (mmcadamsms) [inline smokey/src/main/java/frc/robot/commands/AutoBalanceStepCommand.java:71]: I think we should avoid calling drive twice in the case when it is elapsed.

**Review by mmcadamsms — CHANGES_REQUESTED** [review]
Ok ... I've left comments in the PR take a look and decide how you'd like to proceed.

File: camp/abolotski/UltrasonicSensorTest/src/main/java/frc/robot/commands/GetUltrasonic.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+
```
[the remark this answers is not in the document]
Comment (josieammer) [reply camp/abolotski/UltrasonicSensorTest/src/main/java/frc/robot/commands/GetUltrasonic.java:1]: Not yet.  We could pull it out to a mini project until we need it in Smokey.

File: smokey/src/main/java/frc/robot/subsystems/NavxSubsystem.java [hunk 4]
```diff
@@ -30,7 +36,50 @@ public NavxSubsystem() {}
    * 'forwards' direction.
    */
   public void zeroGyroscope() {
-        swerveNavx.zeroYaw();
+    if(swerveNavx.isCalibrating()){
```
[the remark this answers is not in the document]
Comment (josieammer) [reply smokey/src/main/java/frc/robot/subsystems/NavxSubsystem.java:48]: I think a high-level retry would be a good idea as long as we can guarantee the robot won't move during the wait time.

File: smokey/src/main/java/frc/robot/swerveHelpers/Falcon500SteerControllerFactoryBuilder.java [hunk 5]
```diff
@@ -201,10 +201,10 @@ public void setReferenceAngle(double referenceAngleRadians) {
                 if (++resetIteration >= ENCODER_RESET_ITERATIONS) {
```
Comment (josieammer) [inline smokey/src/main/java/frc/robot/swerveHelpers/Falcon500SteerControllerFactoryBuilder.java:201]: If the robot is enabled before this happens the first time, the robot is disoriented.  
Is there a way to ensure this happens without delay at construction?
If not, should we set a flag on the first time through, and then print the message once to alert the user it's OK to enable.

File: camp/abolotski/UltrasonicSensorTest/src/main/java/frc/robot/commands/GetUltrasonic.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (asherbolotski) [reply camp/abolotski/UltrasonicSensorTest/src/main/java/frc/robot/commands/GetUltrasonic.java:1]: This file is in the camp folder so it doesn't affect smokey

File: smokey/src/main/java/frc/robot/Constants.java [hunk 6]
```diff
@@ -9,6 +9,7 @@
 // ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ ʕ •ᴥ•ʔ ʕ•ᴥ•  ʔ ʕ  •ᴥ•ʔ ʕ •`ᴥ´•ʔ ʕ° •° ʔ 
 
 package frc.robot;
+import java.lang.Math;
```
[the remark this answers is not in the document]
Comment (asherbolotski) [reply smokey/src/main/java/frc/robot/Constants.java:12]: no, deleted

**Review by mmcadamsms — APPROVED** [review]
ok ... lets do this

**Review by mmcadamsms — APPROVED** [review]
ok ... lets do this

**Pull request merged**